updated contract calls page (#701)
multiversx/mx-docs: docs/developers/developer-reference/sc-payments.md

- `self.call_value().single_esdt()` expects a single ESDT transfer, fails otherwise. Will return the received `EsdtTokenPayment`. It is a special case of `multi_esdt`, where `N` is 1.
- `self.call_value().single_fungible_esdt()` further restricts `single_esdt` to only fungible tokens, so those with their nonce zero. Returns the token identifier and amount, as pair.
- `self.call_value().egld_or_single_esdt()` retrieves an object of type `EgldOrEsdtTokenPayment`. Will halt execution in case of ESDT multi-transfer.
- `self.call_value().egld_or_single_fungible_esdt()` further restricts `egld_or_single_esdt` to fungible ESDT tokens. It will return a pair of `EgldOrEsdtTokenIdentifier` and an amount.
- `self.call_value().any_payment()` is the most general payment retriever. Never stops execution. Returns an object of type `EgldOrMultiEsdtPayment`.

---

[comment]: # (mx-context-auto)

## Sending payments

We have seen how contracts can accomodate receiving tokens. Sending them is, in principle, even more straightforward, as it only involves calling methods from the `self.send()` component.

[comment]: # (mx-context-auto)

### Sending payments directly

> Contracts can send tokens directly, without calling any endpoint at the destination, by using several methods from the API.

Reviewer: If the contract is payable by default, otherwise it's disallowed.

Author: This is about sending, not receiving. We talked about direct transfers and payable contracts in a different section.

The following methods will perform a transfer-execute call. This is a type of asynchronous call that does not provide a callback, or any other feedback from the receiver. For direct transferring of funds, this call type is ideal. 

- `self.send().direct_egld(to, amount)` sends EGLD directly to an address.
- `self.send().direct_non_zero_egld(to, amount)` sends EGLD directly to an address, if the amount is non-zero. Does nothing otherwise.
- `self.send().direct(to, token, nonce, amount)` sends EGLD or a single ESDT token directly to an address. The `token` argument is of type `EgldOrEsdtTokenIdentifier`.
- `self.send().direct_non_zero(to, token, nonce, amount)` sends EGLD or a single ESDT token directly to an address, if the amount is non-zero. Does nothing otherwise.
- `self.send().direct_esdt(to, token, nonce, amount)` sends a single ESDT token directly to an address.
- `self.send().direct_non_zero_esdt_payment(to, payment)` sends a single ESDT token directly to an address, if the amount is non-zero. Does nothing otherwise.

> - `self.send().direct_multi(to, payments)` sends one or more ESDT tokens to destination. The `payments` argument is a list of such payments.

Reviewer 2: ```suggestion
- `self.send().direct_multi(to, payments)` sends one or more ESDT tokens to destination. The `payments` argument is a list of such payments. If at least one of the amounts is zero, execution will fail.
```

It is also possible to transfer tokens via an async call:
- `self.send().transfer_esdt_via_async_call(to, token, nonce, amount)`
- `self.send().transfer_esdt_non_zero_via_async_call(to, token, nonce, amount)`
- `self.send().transfer_multiple_esdt_via_async_call(to, payments)`

[comment]: # (mx-context-auto)

### Sending payments to contract endpoints

Sending tokens to a contract endpoint is a contract call, and we have a [type-safe and clean way of doing this](/developers/developer-reference/sc-contract-calls).

Even if you do not know what the endpoint or the arguments will be at compile time, we still recommend creating a `ContractCall` first, and then decorating and sending it, as in [these examples](/developers/developer-reference/sc-contract-calls#no-proxy).

> There are, howver, a few direct API methods. Just to mention them:

Reviewer 2: ```suggestion
However, a few direct API methods exist. Just to mention them:
```
